### PR TITLE
Add cost callback for rest calls

### PR DESCRIPTION
**Pull request by laurentS**
This PR implements a simple mechanism for keeping track of API "costs" using the mechanism in .

### CODE REVIEW

**Comment by ericboucher** [comment]
@laurentS looks like the tests are not passing for Python 3.8

File: pyproject.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ requests = "^2.25.1"
 # singer-sdk = "^0.4.9"
 # For local SDK dev:
 # singer-sdk = {path = "../singer-sdk", develop = true}
-singer-sdk = {git = "https://github.com/oviohub/meltano-sdk.git", rev = "8e505a074c1bca31bdc7bb3e2c8102fe95bcbe05"}
```
Comment (ericboucher) [inline pyproject.toml]: We don't have a pinned version that fits our need yet?

Comment (ericboucher) [reply pyproject.toml]: It would be nice to move back to a version of the upstream repo

Comment (laurentS) [reply pyproject.toml]: I'm planning to go back to a normal version once the sdk PR is merged, but this allows us to test this code early.

**Review by ericboucher — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (ericboucher) [reply pyproject.toml]: Should we wait to merge?

Comment (laurentS) [reply pyproject.toml]: Yes, I would wait until the sdk is officially supporting the cost logging logic, then we can update and merge this. But for now, this branch allows testing and seeing if the result is useful/usable.

Comment (ericboucher) [reply pyproject.toml]: For reference - https://github.com/meltano/sdk/pull/704

Comment (ericboucher) [reply pyproject.toml]: @laurentS the PR above has been merged :)

**Pull request merged**